internal/service/streamconnection/data_source_stream_connection.go

```diff
@@ -0,0 +1,121 @@
+package streamconnection
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/config"
+)
+
+var _ datasource.DataSource = &streamConnectionDS{}
+var _ datasource.DataSourceWithConfigure = &streamConnectionDS{}
+
+func DataSource() datasource.DataSource {
+	return &streamConnectionDS{
+		DSCommon: config.DSCommon{
+			DataSourceName: streamConnectionName,
+		},
+	}
+}
+
+type streamConnectionDS struct {
+	config.DSCommon
+}
+
+func (d *streamConnectionDS) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: DSAttributes(true),
+	}
+}
+
+// DSAttributes returns the attribute definitions for a single stream connection.
+// `withArguments` marks certain attributes as required (for singular data source) or as computed (for plural data source)
+func DSAttributes(withArguments bool) map[string]schema.Attribute {
+	return map[string]schema.Attribute{
+		"id": schema.StringAttribute{
+			Computed: true,
+		},
+		"project_id": schema.StringAttribute{
+			Required: withArguments,
+			Computed: !withArguments,
+		},
+		"instance_name": schema.StringAttribute{
+			Required: withArguments,
+			Computed: !withArguments,
+		},
+		"connection_name": schema.StringAttribute{
+			Required: withArguments,
+			Computed: !withArguments,
+		},
+		"type": schema.StringAttribute{
+			Computed: true,
+		},
+
+		// cluster type specific
+		"cluster_name": schema.StringAttribute{
```

Comment:
i suppose it's in MMS/API but name seems confusing

Reply:
within the context of creating a connection to a cluster I found it okay :) but yeah this is aligning with the API